### PR TITLE
Fix highlight "Code Search" navbar item

### PR DESCRIPTION
**Pull request by erzhtor**
Closes https://github.com/sourcegraph/sourcegraph/issues/21373

- [x] Add integration or unit tests

## Test plan

Case 1 (search page):
- visit `/search?q=test&patternType=regexp`
- code search dropdown in the navigation `should be` highlighted

Case 2 (repo page):
- visit `/github.com/sourcegraph-testing/zap`
- code search dropdown in the navigation `should be` highlighted

Case 3 (repo file page):
- visit `/github.com/sourcegraph-testing/zap/-/blob/README.md`
- code search dropdown in the navigation `should be` highlighted

Case 3 (batch changes page):
- visit `/batch-changes`
- code search dropdown in the navigation `should not be` highlighted

## App preview:

- [Link](https://sg-web-naman-fix-highlighting-code.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-nriklnrukc.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/web/src/integration/search.test.ts [hunk 1]
```diff
@@ -9,6 +9,7 @@ import { Driver, createDriverForTest } from '@sourcegraph/shared/src/testing/dri
 import { afterEachSaveScreenshotIfFailed } from '@sourcegraph/shared/src/testing/screenshotReporter'
 
 import { WebGraphQlOperations } from '../graphql-operations'
+import { PageRoutes } from '../routes.constants'
```
Comment (thenamankumar) [inline client/web/src/integration/search.test.ts:12]: removing this

File: client/web/src/nav/GlobalNavbar.test.tsx [hunk 2]
```diff
@@ -76,4 +78,54 @@ describe('GlobalNavbar', () => {
         )
         expect(asFragment()).toMatchSnapshot()
     })
+
+    describe('Code Search Dropdown', () => {
```
Comment (erzhtor) [inline client/web/src/nav/GlobalNavbar.test.tsx:82]: I believe, this newly added part is rather an integration test, so it should be moved under `client/web/src/integration`. Let me know if you need any help, please.
image

Comment (thenamankumar) [reply client/web/src/nav/GlobalNavbar.test.tsx:82]: Interesting it ran on my local and CI. 

Will move these to `/integration` .

**Comment by erzhtor** [comment]
@thenamankumar, great job! I think it is good to go with merging once fixing CI! 👍

**Review by thenamankumar — APPROVED** [review]

**Pull request merged**